### ISSUE TITLE
Potential solution for poor arrow color reporting

### ISSUE THREAD

**Issue by v11580** [rooklift/nibbler#259]
The engine scores reported need to be normalized for side to move and scaled for a larger range of numbers.
One example of the problem is that Nibbler reports bad moves as excellent in this position: 
rnb1k2r/pp4pp/3bpp2/q2p4/2NP1B2/2P1P3/P1Q2PPP/R3KB1R b KQkq - 1 11

proposed solution:

change line (106 in 81_arrows)  from *.value() to *.cp  so it reads: 		loss = best_info.cp - info_list[i].cp;
add the following (line 107) line loss == Math.abs(loss);

After the line let colour; (line 164 in 81_arrows) 

//  normalize for side to move:
  let blackToMove = hub.tree.node.board.active === "b";
  
if (blackToMove) {
    info_list[i].cp == -info_list[i].cp;
}

//to scale the var "loss" from the raw engine values:
		
					const origThresholds = [0.30, 0.75, 1.25, 1.75]; //  Configure to your liking. They are centipawn values
					
					function scaleInfoValue(value) { 
					value = value * 0.8;
					return value;
					}
				// map the original thresholds using the scaling function
					const scaledThresholds = origThresholds.map(x => scaleInfoValue(x)); 
					
			
	//stet				
	if (info_list[i].move === show_move && config.next_move_unique_colour) {
					colour = config.actual_move_colour;
				} else if (info_list[i].move === show_move && show_move_was_forced) {
					colour = config.terrible_colour;
				} else if (info_list[i].__touched === false) {
					colour = config.terrible_colour;
				} else if (info_list[i] === best_info) {
					colour = config.best_colour;
					
	// plug in the scaledThresholds variables	

	} else if (loss   <= scaledThresholds[0]) {  
        colour = config.good_colour;  //console.log(i + ': '+ loss.toFixed(3))
			} else if (loss <= scaledThresholds[1]) { 
	 colour = "#496539";     //config.bad_colour; lime green
			} else if (loss <= scaledThresholds[2]) { 
			colour = "#937244";  //config.terrible_colour;   // Orange 
			} else if (loss <= scaledThresholds[3]) { 
			colour = "#725050"; // lite red	
			} else {  	
                       colour = "#714040"; // brick red
		    }

**Comment by rooklift** [comment]
I'm not sure what I'm supposed to be seeing here:

![image](https://github.com/rooklift/nibbler/assets/16438795/1a0f8d23-4da0-43fa-9380-f7904bfd4103)

**Comment by v11580** [comment]
![Nibbler_1](https://github.com/rooklift/nibbler/assets/7061947/05b29e9d-0880-44cc-9305-f37b5c53a89a)
The colors supposedly report on the quality of the moves. This is before the fix.

**Comment by v11580** [comment]
![Nibbler_1](https://github.com/rooklift/nibbler/assets/7061947/8b1e9dde-b71e-41b1-880c-35bf9b175d6c)
After the fix, the colors show the actual quality of the moves.

**Comment by rooklift** [comment]
I don't recognise this GUI.

**Comment by v11580** [comment]
![nibbler_0](https://github.com/rooklift/nibbler/assets/7061947/c83486b1-b979-4182-9321-0036ece0dbef)
This is what the colors should look like.

**Comment by v11580** [comment]
> I don't recognise this GUI.

An old version of the GUI is on my profile.  https://github.com/v11580/nibbler-Dark-Mode

**Comment by rooklift** [comment]
Can you submit a screenshot of the problem using the base Nibbler?

**Comment by v11580** [comment]
> Can you submit a screenshot of the problem using the base Nibbler?
If you ask your version to show 3 variations with the arrows, you will probably see that they are all green. If not, then there is no issue and my original post is a waste  of time, for which I apologize in advance.

**Comment by rooklift** [comment]
Is this Stockfish (or similar) or Leela? For Stockfish I can get this:

![image](https://github.com/rooklift/nibbler/assets/16438795/67ed91e5-cbcf-4a22-94ea-25ccc1cc069d)

**Comment by v11580** [comment]
![cats](https://github.com/rooklift/nibbler/assets/7061947/ca44c45e-5467-4454-9b14-498037b86fb5)
I just downloaded the latest version (everything in the app directory) and this is what I get.

**Comment by rooklift** [comment]
Try temporarily removing your config file so it can run with default settings.

**Comment by v11580** [comment]
> Try temporarily removing your config file so it can run with default settings.

I was wondering about that! good point.

**Comment by v11580** [comment]
OK. I installed a fresh version and the color for the bad moves are reported correctly. I must have introduced a change in the code that created a bug. This was all a mistake of my creation.
